pytorch_lightning/callbacks/stochastic_weight_avg.py

```diff
@@ -280,3 +291,44 @@ def avg_fn(
     ) -> torch.FloatTensor:
         """Adapted from https://github.com/pytorch/pytorch/blob/v1.7.1/torch/optim/swa_utils.py#L95-L97."""
         return averaged_model_parameter + (model_parameter - averaged_model_parameter) / (num_averaged + 1)
+
+    def on_save_checkpoint(
+        self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", checkpoint: Dict[str, Any]
+    ) -> dict:
+        checkpoint_data = {
+            "n_averaged": self.n_averaged,
+            "scheduler_step_count": None if self._swa_scheduler is None else self._swa_scheduler._step_count,
+            "average_model_parameters": self._get_average_model_parameters(trainer),
+        }
+        return checkpoint_data
+
+    def on_load_checkpoint(
+        self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", callback_state: Dict[str, Any]
+    ) -> None:
+        if callback_state:
+            self.n_averaged = callback_state["n_averaged"]
+            self._scheduler_step_count = callback_state["scheduler_step_count"]
+            self._load_average_model_parameters(callback_state["average_model_parameters"])
+        else:
+            rank_zero_warn(
```

Comment:
I believe this should raise a MisConfigurationError. Do you see a use-case where a user would drop the callback state with SWA ?

Reply:
I had imagined this might happen if upgrading the version of PyTorch lightning and then resuming from a checkpoint saved with a previous version. Although I'm not sure that's something that's supposed to be supported so I'd be happy to changes this to an exception.

Reply:
I believe it would be safer to use a `MisConfigurationError` and re-investigate this later on.

Reply:
I tried testing this but realised this method will never be called if the state isn't present in the checkpoint as the trainer already checks that the state is truthy before calling `on_load_checkpoint`: https://github.com/PyTorchLightning/pytorch-lightning/blob/8bf7f9cce744bbc05ebce03bfd0567e3bc414128/pytorch_lightning/trainer/callback_hook.py#L290-L292
So I've just removed the check here.